### PR TITLE
Add ToplogyListeners in gateway

### LINKED ISSUE #12387: Allow listening for updates in BrokerTopologyListener
**Description**

In order for the job push' `ClientStreamService` to detect brokers being added/removed, we need a way to get this information via the topology. We could use a plain membership listener, but that one is unaware of whether a node is a broker or a gateway, whereas the `BrokerTopologyManager` can already provide this information.

We should add the following capabilities:

- [ ] Add a new listener which gets notified when a broker is added or removed
- [ ] When the listener is initially added, it is also initialized the current list of known brokers, to avoid race conditions
- [ ] The listener can be removed via its identity

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR enabled adding topology listeners in gateway. This is required for ClientStreamService.

In addition, this contains following refactoring:
- Moved TopologyUpdateTest to junit5 and replace ActorSchedulerRule with ControlledActorSchedulerExtension. This helps to remove non-deterministic waiting with Awaitility. https://github.com/camunda/zeebe/pull/12447/commits/70a0496a6b0ab09a4529b02b648cce26ef5b386a 
- https://github.com/camunda/zeebe/pull/12447/commits/0bb1ff861974c3357ab691a20bbebc3e8001c84e removes periodic checking of membership state. This is unnecessary because we can rely on membership service to notify when a broker is added or removed. We only have to check once after the startup, to ensure that the membership updates occurred before the actor is started is not lost.
- The above change resulted in flaky tests in `BrokerClientTest` because the fake topology set by the test is overwritten by the topology manager. The tests did not fail before because there was a delay of 5 seconds when topology manager checks for missing events. This is fixed in https://github.com/camunda/zeebe/pull/12447/commits/0945c3088629e5a72b260778c104418afb113b08 which allows waiting until the broker client is fully started. This is an ugly solution. Happy to discuss it. 

## Related issues

closes #12387 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/BrokerClientImpl.java [hunk 1]
```diff
@@ -62,10 +59,11 @@ public BrokerClientImpl(
...
   @Override
-  public void start() {
-    schedulingService.submitActor(topologyManager);
-    schedulingService.submitActor(atomixTransportAdapter);
-    schedulingService.submitActor(requestManager);
+  public Collection<ActorFuture<Void>> start() {
+    final var topologyManagerStarted = topologyManager.start(schedulingService);
+    final var transportStarted = schedulingService.submitActor(atomixTransportAdapter);
+    final var requestManagerStarted = schedulingService.submitActor(requestManager);
+    return List.of(topologyManagerStarted, transportStarted, requestManagerStarted);
```
Comment (npepinpe) [inline gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/BrokerClientImpl.java:66]: :clown_face: If only we could chain futures without having an actor :(

**Review by npepinpe — APPROVED** [review]
:rocket: Thanks! I wouldn't worry too much about ugly solutions or the likes. If we were to invest more into the broker client/broker topology manager, I would propose following Chris' idea and moving towards reusing the a unified topology manager instead of two separate implementations.

**Comment by deepthidevaki** [comment]
bors merge

**Comment by deepthidevaki** [comment]
bors retry

**Pull request merged**